tests/operations/unary/RandomBool.cpp

```diff
@@ -1,11 +1,63 @@
 #define CATCH_CONFIG_MAIN
 #include "Catch/single_include/catch2/catch.hpp"
 
+#include "conduit/include/uitsl/polyfill/bit_cast.hpp"
+
+#include "Empirical/include/emp/data/DataNode.hpp"
+
+#include "sgpl/algorithm/execute_cpu.hpp"
+#include "sgpl/hardware/Cpu.hpp"
 #include "sgpl/operations/unary/RandomBool.hpp"
+#include "sgpl/program/Program.hpp"
+#include "sgpl/spec/Spec.hpp"
+#include "sgpl/utility/ThreadLocalRandom.hpp"
+
+using library_t = sgpl::OpLibrary<sgpl::RandomBool>;
+struct spec_t : public sgpl::Spec<library_t>{
+  // this is here so that we can step through the operations properly
+  static constexpr inline size_t switch_steps{ 1 }; // eslint-disable-line no-eval
+  // at least 20 cores are required
+  static constexpr inline size_t num_cores{ 20 }; // eslint-disable-line no-eval
+};
 
 TEST_CASE("Test RandomBool") {
```

Comment:
This test is more complicated, add a high level comment above describing what we’re testing and how